### PR TITLE
backport: bitcoin#10583 - [RPC] Split part of validateaddress into getaddressinfo

### PR DESCRIPTION
**Pull request by xdustinface**

### CODE REVIEW

File: src/rpc/misc.cpp [hunk 1]
```diff
@@ -114,27 +114,6 @@ UniValue mnsync(const JSONRPCRequest& request)
     return "failure";
```
Comment (PastaPastaPasta) [inline src/rpc/misc.cpp:114]: 3bd6522c0a18f37ef0ce4e9918bd5530c60019fb missing changes in makefile.am?

File: src/Makefile.am [hunk 2]
```diff
@@ -537,6 +536,7 @@ libdash_common_a_SOURCES = \
   protocol.cpp \
   saltedhasher.cpp \
   scheduler.cpp \
+  script/ismine.cpp \
```
Comment (PastaPastaPasta) [inline src/Makefile.am:539]: Hmm, looks like you didn't include makefile.am changes in 3bd6522c0a18f37ef0ce4e9918bd5530c60019fb due to them being reverted in 96ddfa5fb22022e1f8c5fbb1f197b47b095d6af0

**Review by PastaPastaPasta — APPROVED** [review]
Looks good, utACK

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -3569,6 +3570,147 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
...
+            "      ,...\n"
+            "    ]\n"
+            "  \"sigsrequired\" : xxxxx        (numeric, optional) Number of signatures required to spend multisig output (only if \"script\" is \"multisig\")\n"
+            "  \"pubkey\" : \"publickeyhex\",    (string, optional) The hex value of the raw public key, for single-key addresses (possibly embedded in P2SH)\n"
+            "  \"embedded\" : {...},           (object, optional) Information about the address embedded in P2SH, if relevant and known. It includes all getaddressinfo output fields for the embedded address, excluding metadata (\"timestamp\", \"hdkeypath\") and relation to the wallet (\"ismine\", \"iswatchonly\", \"account\").\n"
+            "  \"iscompressed\" : true|false,  (boolean) If the address is compressed\n"
+            "  \"account\" : \"account\"         (string) The account associated with the address, \"\" is the default account\n"
+            "  \"timestamp\" : timestamp,      (number, optional) The creation time of the key if available in seconds since epoch (Jan 1 1970 GMT)\n"
+            "  \"hdkeypath\" : \"keypath\"       (string, optional) The HD keypath if the key is HD and available\n"
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3657]: `hdchainid` help string is missing

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp (hunk 3 above)
Comment (xdustinface) [reply src/wallet/rpcwallet.cpp:3657]: Added in 8afca64

File: src/wallet/rpcwallet.cpp [hunk 4]
```diff
@@ -3569,6 +3570,148 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
...
+            "  \"hdchainid\" : \"<hash160>\"     (string, optional) The Hash160 of the hd chain id\n"
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3658]:
```suggestion
            "  \"hdchainid\" : \"<hash>\"        (string, optional) The ID of the HD chain\n"
```

Comment (xdustinface) [reply src/wallet/rpcwallet.cpp]: ahh, i see.. the previous one 🙈

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
lgtm, utACK

waiting on CI

**Pull request merged**